src/app/clusters/bindings/BindingManager.cpp

```diff
@@ -204,6 +218,7 @@ CHIP_ERROR BindingManager::NotifyBoundClusterChanged(EndpointId endpoint, Cluste
                 {
                     // We already have an active connection
                     mBoundDeviceChangedHandler(*iter, peerDevice, context);
+                    mBoundDeviceContextReleaseHandler(context);
```

Comment:
ReleaseHandler should be called only once per context, so you should probably call it after the for-loop (unless the context was added to the pending notification list).